pandas/core/dtypes/cast.py

```diff
@@ -1207,6 +1206,19 @@ def astype_nansafe(
     return arr.astype(dtype, copy=copy)
 
 
+def astype_float_to_int_nansafe(
```

Comment:
this is already done in astype_nansafe and IIRC in Index constructor as well. can you consolidate.

Reply:
> this is already done in astype_nansafe

yes, this changes the call in astype_nansafe to use this new function, so that we can use this lighter-weight func in construct_1d_ndarray_preserving_na

> and IIRC in Index constructor as well. can you consolidate.

similar but not identical (we also do it a third way for IntegerArray).  some combination of #40489 and #40110 i think will make sharing feasible

Reply:
kk great. ok happy to merge here and do followsup.